src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -3455,12 +3455,7 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
 
             isOnSystem32ModulePath = true;
 
-            if (BaseSkipEditionCheck)
-            {
-                return true;
-            }
-
-            return Utils.IsPSEditionSupported(compatiblePSEditions);
+            return BaseSkipEditionCheck || Utils.IsPSEditionSupported(compatiblePSEditions);
```

Comment:
I wonder that we move the block - have we I side effect in lines 3450-3451?

Reply:
The `if` block? My goal is for the `PSModuleInfo` to accurately reflect the module manifest being loaded. It seems to be working for our purposes at present -- I moved the block down because I realised `isOnSystem32ModulePath` wasn't always set when a module was actually found there.